irit_rst_dt/config/common.py

```diff
@@ -0,0 +1,118 @@
+"""Commonly used configuration options"""
+
+from collections import namedtuple
+import six
+# from attelo.decoding.astar import (AstarArgs,
+#                                    AstarDecoder,
+#                                    Heuristic,
+#                                    RfcConstraint)
+from attelo.decoding.baseline import (LastBaseline,
+                                      LocalBaseline)
+from attelo.harness.config import (EvaluationConfig,
+                                   LearnerConfig,
+                                   Keyed)
+from attelo.learning.oracle import (AttachOracle, LabelOracle)
+from attelo.parser.full import (JointPipeline,
+                                PostlabelPipeline)
+
+
+def combined_key(*variants):
+    """return a key from a list of objects that have a
+    `key` field each"""
+    return '-'.join(v if isinstance(v, six.string_types) else v.key
+                    for v in variants)
+
+
+Settings = namedtuple('Settings',
+                      ['key', 'intra', 'oracle', 'children'])
+"""
+Note that this is subclass of Keyed
```

Comment:
It has a key. Not a subclass, sorry, brain fart. Corrected doc string
should talk about just the fact that it has a key and can thus be used for
combined keys. Type error in my docs, sheesh.
On Fri, 29 May 2015 at 14:23, Mathieu Morey <notifications@github.com>
wrote:

> In irit_rst_dt/config/common.py
> <https://github.com/irit-melodi/irit-rst-dt/pull/20#discussion_r31319536>:
>
> > +from attelo.learning.oracle import (AttachOracle, LabelOracle)
> > +from attelo.parser.full import (JointPipeline,
> > +                                PostlabelPipeline)
> > +
> > +
> > +def combined_key(*variants):
> > +    """return a key from a list of objects that have a
> > +    `key` field each"""
> > +    return '-'.join(v if isinstance(v, six.string_types) else v.key
> > +                    for v in variants)
> > +
> > +
> > +Settings = namedtuple('Settings',
> > +                      ['key', 'intra', 'oracle', 'children'])
> > +"""
> > +Note that this is subclass of Keyed
>
> I am not sure I see how this is a subclass of Keyed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/irit-melodi/irit-rst-dt/pull/20/files#r31319536>.
>